### PR TITLE
Add docker-compose.

### PR DESCRIPTION
**Pull request by thinking-tower**
# Expected outputs

- localhost:8000 return `404 not found`, meaning that the Go app is running.
- localhost:3000 returns the PupilFirst page, meaning that the React app is running.

# References
- [Install and run Postgres with an extension using docker-compose](https://gist.github.com/leopoldodonnell/b0b7e06943bd389560184d948bdc2d5b)
- [CompileDaemon](https://github.com/githubnemo/CompileDaemon)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        networks:
+            - pupilgo
+    redis:
+        image: "redis:alpine"
+        restart: unless-stopped
+        networks:
+            - pupilgo
+    backend:
+        build: .
+        command: go run cmd/api/main.go
```
Comment (cbrom) [inline docker-compose.yml]: I think we should make GraphQL our main API so change this to `go run cmd/graphql/main.go`

File: postgres/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM postgres:12.3
+
+ENV POSTGRES_DB 'go_lms'
+ENV POSTGRES_USER 'postgres'
+ENV POSTGRES_PASSWORD '1772'
```
Comment (cbrom) [inline postgres/Dockerfile]: Can you change these to .env file? It's manageable and easy for debugging.
like so 
`env_file: `
 `          - ".env"`

File: .env [hunk 3]
```diff
@@ -1,8 +1,9 @@
-STORAGE_HOST=localhost
+STORAGE_HOST=db
```
Comment (cbrom) [inline .env]: I don't get why this one changed. Is it because the postgres container is named `db`?

**Review by cbrom — CHANGES_REQUESTED** [review]

File: .env (hunk 3 above)
Comment (thinking-tower) [reply .env]: When you're inside of the go container, localhost:5432 doesn't point to the postgres container. You have to use what I called the postgres container (db) docker-compose.yml file. However if you're not using docker-compose then yes this shouldn't be changed.

**Pull request merged**